src/test/java/com/mathworks/ci/systemtests/TestResultVisualizationIT.java

```diff
@@ -0,0 +1,340 @@
+        project.setScm(new ExtractResourceSCM(Utilities.getURLForTestData()));
+
+        UseMatlabVersionBuildWrapper buildWrapper = new UseMatlabVersionBuildWrapper();
+        buildWrapper.setMatlabBuildWrapperContent(new MatlabBuildWrapperContent(Message.getValue("matlab.custom.location"), Utilities.getMatlabRoot()));
+        project.getBuildWrappersList().add(buildWrapper);
+
+        // Run tests through Run Build step
+        RunMatlabBuildBuilder buildtoolBuilder = new RunMatlabBuildBuilder();
+        buildtoolBuilder.setTasks("test");
+        project.getBuildersList().add(buildtoolBuilder);
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        String hyperlinkInSummary = getHyperlinkFromBuildStatus(build);
+
+        String hyperlinkOfTestResultsTab = getTestResultTabLinkFromSidePanel(build);
+
+        assertTrue(hyperlinkOfTestResultsTab.contains(hyperlinkInSummary));
+    }
+
+    @Test
+    public void verifyContentInTestResultsTable() throws Exception{
+        FreeStyleProject project = jenkins.createFreeStyleProject();
+        project.setScm(new ExtractResourceSCM(Utilities.getURLForTestData()));
+
+        UseMatlabVersionBuildWrapper buildWrapper = new UseMatlabVersionBuildWrapper();
+        buildWrapper.setMatlabBuildWrapperContent(new MatlabBuildWrapperContent(Message.getValue("matlab.custom.location"), Utilities.getMatlabRoot()));
+        project.getBuildWrappersList().add(buildWrapper);
+
+        // Run tests through Run Build step
+        RunMatlabBuildBuilder buildtoolBuilder = new RunMatlabBuildBuilder();
+        buildtoolBuilder.setTasks("test");
+        project.getBuildersList().add(buildtoolBuilder);
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        // Verify button text
+        String totalTestsButtonText = getButtonText("TOTAL", build);
+        assertTrue(totalTestsButtonText.contains("4"));
+        assertTrue(totalTestsButtonText.contains("Total Tests"));
+
+        String passedButtonText = getButtonText("PASSED", build);
+        assertTrue(passedButtonText.contains("1"));
+        assertTrue(passedButtonText.contains("Passed"));
+
+        String failedButtonText = getButtonText("FAILED", build);
+        assertTrue(failedButtonText.contains("3"));
+        assertTrue(failedButtonText.contains("Failed"));
+
+        String incompleteButtonText = getButtonText("INCOMPLETE", build);
+        assertTrue(incompleteButtonText.contains("0"));
+        assertTrue(incompleteButtonText.contains("Incomplete"));
+
+        String notRunButtonText = getButtonText("NOT_RUN", build);
+        assertTrue(notRunButtonText.contains("0"));
+        assertTrue(notRunButtonText.contains("Not Run"));
+
+        // Verify the test filenames shown
+        String[] expectedTestFiles = {" testMultiply ", " squareTest ", " testSquare ", " testSum "};
+        String[] testFiles = getTestFilesInTable(build);
+        assertEquals(testFiles, expectedTestFiles);
```

Comment:
Should we also add tests:
- to verify if the test file status is correct?
- to verify the help text that appears when we hover above the test file name?

Reply:
As there are unit tests for checking the test file and tests status, we need not add a system test for this. Same for the text(shown when hovering) as well